### ISSUE TITLE
AJax Chat use myBB avatars

### ISSUE THREAD

**Issue by icolinmeiklem** [Frug/AJAX-Chat#209]
i want to know how to code mybb avatars into ajax chat so that peoples names on ajax chat show there mybb avatar along side them.

**Comment by patmaclondon** [comment]
It’s not perfect yet with my code, with avatars, but i did get the Names and and ID’s integrated from MyBB into Ajax chat, that work seemly. However what I stated before is I just haven’t had the time. My code works as followed following MySQL

MyBB handles all the user accounts, it handles all of it, the way I see it is that AJAX chat has a problem with talking to the database properly, this can be because I haven’t written the code correctly, but why I’m trying to say is that I have it so far that the database from myBB works fine with ajax chat. My future plan and 1st on the list is to integrate the avatars. I would like nothing more then that would work, but it is a SQL problem that has to be solved, the data integration between ajax and mybb is already there in my code, just have to figure out how to call the avatar in ajax, hope that makes sense.

Long story short, I’m getting there, but programming is very time consuming….

**Comment by patmaclondon** [comment]
It’s not perfect yet with my code, with avatars, but i did get the Names and and ID’s integrated from MyBB into Ajax chat, that work seemly. However what I stated before is I just haven’t had the time. My code works as followed following MySQL

MyBB handles all the user accounts, it handles all of it, the way I see it is that AJAX chat has a problem with talking to the database properly, this can be because I haven’t written the code correctly, but why I’m trying to say is that I have it so far that the database from myBB works fine with ajax chat. My future plan and 1st on the list is to integrate the avatars. I would like nothing more then that would work, but it is a SQL problem that has to be solved, the data integration between ajax and mybb is already there in my code, just have to figure out how to call the avatar in ajax, hope that makes sense.

Long story short, I’m getting there, but programming is very time consuming….

One last thing, PHPBB should handle it the same way, but it the PHP has to be written for that forum software, I still think the load time and the overkill of PHPBB is too much, same as wordpress for a simple website, myBB Loads faster and doesn't have that much crap to go through as PHPBB

**Comment by icolinmeiklem** [comment]
ok just i have seen ajax chat scripts that people have edited and have avatars when do you think the release will be im willing to pay for a seperate one .

**Comment by patmaclondon** [comment]
My Ajax chat has avtatars as well, BUT i does it though a group ID something like "if ID = 0 show blabla.jpg" what you are asking can't be very difficult but it needs to access the data from the SQL database rather then getting it static based on the group ID. If it already gets the name out of MyBB of PHPBB my guess is it is one or 2 lines of code.

**Comment by icolinmeiklem** [comment]
so i could custom put the avatar in the id? so it shows it on the ajax chat i just want to be able to put avatars next to people names i dont care if i have to manually put the image into an id or whatever. it doesnt have to link to the mybb one just aslong as i can put one next to there name on ajax if this makes sense?

**Comment by patmaclondon** [comment]
What I'm doing now is installing the english version of my chat for you, you have to understand that with programming you have to be a bit patient, but I will have a look. I'll let you know when that is up and running, and if you could have a look at the site and point out exactly what you are after (Even though I think i have a good idea) we can take it from there.

**Comment by icolinmeiklem** [comment]
this is what i am after http://i.imgur.com/QOwRsJp.png just the team logo idea not the other image next to it and my site is pesrevolution.com

**Comment by patmaclondon** [comment]
![qowrsjp](https://cloud.githubusercontent.com/assets/3041841/9489567/e260d6f2-4bd9-11e5-8e50-56076fc7951f.jpg)

**Comment by icolinmeiklem** [comment]
its a fifa site that just means they dont have a team i dont want that bit i just want to put the crest logo next to it

**Comment by patmaclondon** [comment]
So you just want that picture next to the name? and that picture is uploaded in PHPBB or MyBB is that correct? So when you go back to the forum and in your user account settings that is where you change it....?

**Comment by icolinmeiklem** [comment]
yes thats what i want.

**Comment by patmaclondon** [comment]
OK I think I have a pretty good idea how to do that, like I said, give a liitle bit of time, I'm setting up a dummy chat site at the moment, so if you want I can ven make you admin on there, just for test purpose, so when I figured it out it will be backed up, so you can't do anything wrong to screw it up on the site... I'm quite interested in this, so it will help me as well to get this working.

**Comment by icolinmeiklem** [comment]
FANTASTIC yeah that would be great to be made admin to test purpose. i know this is abit rude to ask as its your personal time but do you know how long?

**Comment by patmaclondon** [comment]
I'm setting up the site as we speak, but I have to integrate a backup, give me 30 minutes or something to get the site live, I have to install MyBB first and then upload/replace the files... Not to be rude from my end, but I can't go faster then my FTP connection

**Comment by icolinmeiklem** [comment]
ok thanks a lot mate!!

**Comment by patmaclondon** [comment]
Another thing I haven't been able to do yet and I would really like to write in the future is Private chat to someone in a seperate window, so you can still be on the chat, and have a private conversation with someone at the same time

**Comment by icolinmeiklem** [comment]
that would be a fantastic feature also!

**Comment by patmaclondon** [comment]
What I don't like is the MyBB and PHP setup, there is another forum software called mini BB, it does almost nothing, so I'm looking into that, what I'm trying to say is your whole site will speed up, if that is handled What I mean by that is, it's not full of crap like themes and stuff, I find it quite hard to program in MyBB, because it is all webbased, I do everything in Dreamweaver, and I like to understand the code, both MyBB and PHPBB are so big, I spend more time searching through code that acyually doing work

**Comment by icolinmeiklem** [comment]
yeah i have heard of it but it would make my website look dry as i am only having a forum based website

**Comment by patmaclondon** [comment]
In an ideal world I would like to have one package, so no seperate ajax chat, just create database, install the package and bam chat is there integrated, but the danger is that I'm all alone then, and have to start a seperate GITHUB forum or something, I already had loads of interest about my chat, but it is of course modified code, it is still based on AJAX chat, with the floating windows, graphics ect I already took a risk by stirring away from the community sort of speak, does that make sense

**Comment by icolinmeiklem** [comment]
yeah i think i know what you mean i get total confused when it comes to coding i was wanting a site for standalone chat instead of a forum but i cant code websites i can only use mybb so that ideas kind of out the window for me

**Comment by patmaclondon** [comment]
There is a stand alone version of Ajax chat but you loose a lot of the functions and control. it is better to have it integrated with a forum.

**Comment by patmaclondon** [comment]
How long has your site been up?

**Comment by icolinmeiklem** [comment]
i just started it few days ago i plan on launching by 1st september as that is when the game the site is about is launched. thats why i am kind of desperate for this avatars thing

**Comment by patmaclondon** [comment]
Is it a football thing?

**Comment by icolinmeiklem** [comment]
yeah its for pes 2016 the football game so people can play eachother in leagues make transfers with other teams ect

**Comment by patmaclondon** [comment]
Owww ok.... well at the end of the day, a forum is usefull to have in that case, you also might want to think about putting adverts on the site, you will get paid when people click on them, so you get your money back you invest in the site yourself

**Comment by icolinmeiklem** [comment]
yeah im making this chat thing my main focus atm then i can think of that kind of stuff then the members the hard part.

**Comment by patmaclondon** [comment]
I've been doing this on the side as a hobby for more then 12 years now, member is a thing of seo, and backlinks, also a lot of programmers are technical, so if it works jippie, but it also has to do work and look the way users want, if not they won't stay

**Comment by icolinmeiklem** [comment]
yeah in the football gaming forum its a competitive market there is a good amount of sites for it so i want it looking as pro as i can with many features

**Comment by patmaclondon** [comment]
You need to think about a couple of things here, first of all when you grow, you will need moderators, getting members is one thing, kepping them is another, starting a new site is neither good nor bad, but bad reviews are bad. So you have to have some sort of order in place. So your admin and organisation has to be in order

**Comment by icolinmeiklem** [comment]
yeah i have another administrator helping me and i am in touch with a fifa based website so we are linking together as its total diff games so we dont call it rival against each site so to speak

**Comment by patmaclondon** [comment]
The biggest challenge you will face is the chat can not be empty, other wise someone will login and he is not going to talk to himself, it's a circel, I've no one is there no one will come... it's very hard to crack into that
Once you have loyal member being there, and there is someone, then the ball gets rolling

**Comment by icolinmeiklem** [comment]
yeah i have a fair amount of people willing to join so as long as i get all this sorted before release of the game then i should be sitting in a not bad position

**Comment by patmaclondon** [comment]
OK, you must be thinking by now, why is this guy helping me, really simple, I don't have the time anymore to run a chat site, even when I had the site I was never interested what was going on on there, but I am very interested in the code behind it ect, furthermore I do this for a living, it's part of what my company does. so if I figure a problem out, and the code is fine, I can use that for other projects. I just don't want to have my own chat site, but I do like being parts of one that someone else runs manages, and be a technical adviser, so I don't have to deal with the users, or the content of the site

**Comment by gWorldz** [comment]
+1 (Y)

On 8/26/15, Patrick Smit <notifications@github.com> wrote:
> OK, you must be thinking by now, why is this guy helping me, really simple,
> I don't have the time anymore to run a chat site, even when I had the site I
> was never interested what was going on on there, but I am very interested in
> the code behind it ect, furthermore I do this for a living, it's part of
> what my company does. so if I figure a problem out, and the code is fine, I
> can use that for other projects. I just don't want to have my own chat site,
> but I do like being parts of one that someone else runs manages, and be a
> technical adviser, so I don't have to deal with the users, or the content of
> the site
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/Frug/AJAX-Chat/issues/209#issuecomment-134935875


-- 


   - ===[ SIGNATURE ]=====================
   * Name:* Doc gWorldz | ~cXc~
   * Email:* Personal
   <https://mail.google.com/mail/u/0/?view=cm&fs=1&tf=1&to=chaosxonline@gmail.com>
|
   Facebook
   <https://mail.google.com/mail/u/0/?view=cm&fs=1&tf=1&to=gworldz@facebook.com>
|
   Business
   <https://mail.google.com/mail/u/0/?view=cm&fs=1&tf=1&to=admin@gworldz.com>

   * Website:* http://gworldz.com
* Social Links:*  <https://www.facebook.com/gWorldz>
   <https://twitter.com/gWorldz>
   <https://www.linkedin.com/pub/doc-g-worldz/14/721/372>
   <http://gworldz.wordpress.com/> * <http://gworldz.deviantart.com/>*
   <http://gplus.to/gWorldz>
   <https://mail.google.com/mail/u/0/?view=cm&fs=1&tf=1&to=email2chaos@yahoo.com>
     <https://www.youtube.com/channel/UCPasmY51U26VW1k7lbzLi7A>
   <http://gworldz.com/feed/>
   ==================[ END SIGNATURE ]===

**Comment by patmaclondon** [comment]
Site is still copying by the way, also you are in luck, I'm waiting for documents from a client, so I have not much to do at the moment

**Comment by icolinmeiklem** [comment]
lol you can help as much as you like code wise and super looking forward to seeing if it works out

**Comment by patmaclondon** [comment]
Are you UK based?

**Comment by icolinmeiklem** [comment]
yes im uk based

**Comment by patmaclondon** [comment]
Cool that makes life a lot easier, they have released a lot of new laws on internet hosting ect. the last couple of years, I'm UK based as well so I know all the rules and regulations, they differ in different counties, also I don't have to worry about implementing US english, ect.

**Comment by icolinmeiklem** [comment]
i dont know much when it comes to all that stuff but i know all the stuff i am using is legal aint no priate stuff or anythinglike that

**Comment by patmaclondon** [comment]
Yeah but when your website gets more members, you have to understand not all of them are into football, some of them are interested in screwing other people , make life difficult for other members, promoting kiddy porn, (Sounds hard but it happens)...

**Comment by icolinmeiklem** [comment]
yeah i know what you mean i will be highly monitoring the forum and they will be banned instantly

**Comment by patmaclondon** [comment]
Also make sure you have a proper disclaimer, you just have to cover yourself from all angles, so you are not resposible for the content other people say type or post
Nobody reads the disclaimer, but by law if people have to agree with the terms and conditions you can not be held accountable, another thing is the internet is global, so not one law. it can be tricky, but make sure you cover yourself by UK law

**Comment by icolinmeiklem** [comment]
yeah im going to get the disclaimer and that all sorted out before i release

**Comment by patmaclondon** [comment]
Not saying it will happen to you, but I had some horrible horrible stuff to deal with when I had the chat site, it's only a minority, but they're out there, I had to go to the Police twice, had a divorce happen live on chat, so prepare yourself for some weird crap

**Comment by icolinmeiklem** [comment]
oh that sounds very rare ill just have to keep fingers crossed but ill make sure its well monitored. hows the upload getitng on?

**Comment by pl4sm4** [comment]
Stop spam pls.

**Comment by patmaclondon** [comment]
Do apologise, only now realised it was for public view, also it's not spam it's called flooding, you can delete the post if you like

**Comment by icolinmeiklem** [comment]
are you on skype or anything?

**Comment by patmaclondon** [comment]
Yeah just go to http://patmac.co.uk all my details are there

**Comment by icolinmeiklem** [comment]
great couldnt find skype contacted you on facebook instead
